polynote-frontend/polynote/ui.js

```diff
@@ -964,6 +967,11 @@ export class NotebookCellsUI extends UIEventTarget {
             this.el.replaceChild(newCell.container, currentCell.container);
             currentCell.dispose();
             this.setupCell(newCell);
+            const clientInterpreter = clientInterpreters[language];
+            if (clientInterpreter && clientInterpreter.highlightLanguage && clientInterpreter.highlightLanguage !== language) {
```

Comment:
This logic can go away if we just add a new `vega` language of our own specification